src/components/mc/cuda/mc_cuda.c

```diff
@@ -218,11 +223,37 @@ static void ucc_mc_cuda_chunk_cleanup(ucc_mpool_t *mp, void *obj) //NOLINT: mp i
     }
 }
 
+static void ucc_mc_cuda_chunk_init_managed(ucc_mpool_t *mp, //NOLINT
+                                           void *obj, void *chunk) //NOLINT
+{
+    ucc_mc_buffer_header_t *h = (ucc_mc_buffer_header_t *)obj;
+    cudaError_t             st;
+
+    st = cudaMallocManaged(&h->addr, MC_CUDA_CONFIG->mpool_elem_size, cudaMemAttachGlobal);
+    if (st != cudaSuccess) {
+        // h->addr will be 0 so ucc_mc_cuda_mem_alloc_pool function will
```

Comment:
Is it guaranteed? I can't find this in the doc. For safety, we could manually set `h->addr` to `NULL` here